tests/crypto/test_cases/test_hkdf.c

```diff
@@ -0,0 +1,148 @@
+
+#include "common_test.h"
+#include <mbedtls/hkdf.h>
+
+LOG_MODULE_REGISTER(test_hkdf, LOG_LEVEL_INF);
+
+extern test_vector_hkdf_t __start_test_vector_hkdf_data[];
+extern test_vector_hkdf_t __stop_test_vector_hkdf_data[];
+
+/* TODO: Possibly tune buffers which have lower size requirements */
+#define HKDF_BUF_SIZE (258)
+
+/* Input key material */
+static uint8_t m_hkdf_ikm_buf[HKDF_BUF_SIZE];
+/* Output key material */
+static uint8_t m_hkdf_okm_buf[HKDF_BUF_SIZE];
+/* Pseudo-random key (material) */
+static uint8_t m_hkdf_prk_buf[HKDF_BUF_SIZE];
+static uint8_t m_hkdf_salt_buf[HKDF_BUF_SIZE];
+static uint8_t m_hkdf_info_buf[HKDF_BUF_SIZE];
+static uint8_t m_hkdf_expected_okm_buf[HKDF_BUF_SIZE];
+
+static uint8_t *p_hkdp_salt;
+static uint8_t *p_hkdp_info;
+
+static test_vector_hkdf_t *p_test_vector;
+
+static size_t ikm_len;
+static size_t okm_len;
+static size_t prk_len;
+static size_t salt_len;
+static size_t info_len;
+static size_t expected_okm_len;
+
+void hkdf_clear_buffers(void)
+{
+	memset(m_hkdf_ikm_buf, 0x00, sizeof(m_hkdf_ikm_buf));
+	memset(m_hkdf_okm_buf, 0xFF, sizeof(m_hkdf_okm_buf));
+	memset(m_hkdf_prk_buf, 0x00, sizeof(m_hkdf_prk_buf));
+	memset(m_hkdf_salt_buf, 0x00, sizeof(m_hkdf_salt_buf));
+	memset(m_hkdf_info_buf, 0x00, sizeof(m_hkdf_info_buf));
+	memset(m_hkdf_expected_okm_buf, 0x00, sizeof(m_hkdf_expected_okm_buf));
+}
+
+__attribute__((noinline)) void unhexify_hkdf(void)
+{
+	/* Fetch and unhexify test vectors. */
+	ikm_len = hex2bin(p_test_vector->p_ikm, strlen(p_test_vector->p_ikm),
+			  m_hkdf_ikm_buf, strlen(p_test_vector->p_ikm));
+	prk_len = hex2bin(p_test_vector->p_prk, strlen(p_test_vector->p_prk),
+			  m_hkdf_prk_buf, strlen(p_test_vector->p_prk));
+	salt_len = hex2bin(p_test_vector->p_salt, strlen(p_test_vector->p_salt),
+			   m_hkdf_salt_buf, strlen(p_test_vector->p_salt));
+	info_len = hex2bin(p_test_vector->p_info, strlen(p_test_vector->p_info),
+			   m_hkdf_info_buf, strlen(p_test_vector->p_info));
+	expected_okm_len =
+		hex2bin(p_test_vector->p_okm, strlen(p_test_vector->p_okm),
+			m_hkdf_expected_okm_buf, strlen(p_test_vector->p_okm));
+	okm_len = expected_okm_len;
+
+	p_hkdp_salt = (salt_len == 0) ? NULL : m_hkdf_salt_buf;
+	p_hkdp_info = (info_len == 0) ? NULL : m_hkdf_info_buf;
+}
+
+void hkdf_setup(void)
+{
+	static uint32_t i;
+
+	hkdf_clear_buffers();
+	p_test_vector =
+		ITEM_GET(test_vector_hkdf_data, test_vector_hkdf_t, i++);
+	unhexify_hkdf();
+}
+
+/**@brief Function for the test execution.
+ */
+void exec_test_case_hkdf(void)
+{
+	int err_code = -1;
+
+	/* Calculation of the HKDF extract and expand. */
+	start_time_measurement();
+
+	const mbedtls_md_info_t *p_md_info =
+		mbedtls_md_info_from_type(p_test_vector->digest_type);
+	err_code = mbedtls_hkdf(p_md_info, p_hkdp_salt, salt_len,
+				m_hkdf_ikm_buf, ikm_len, m_hkdf_info_buf,
+				info_len, m_hkdf_okm_buf, okm_len);
+	stop_time_measurement();
+
+	LOG_DBG("Error code extract and expand: %d, expected: %d", err_code,
```

Comment:
Probably better handled with a LOG_ON_ERROR (assert-like) statement to not influence the result even in debug